src/Chapter2.hs

```diff
@@ -857,7 +886,10 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind list = case list of
+  [] -> []
+  _ -> last list : rewind (init list)
```

Comment:
Your solution is correct! 👏🏼 
However, it is slow. 
In lists, it is quite slow to add or take anything at the end of the list. That is why it is always better to rewrite it with the `:` cons. Remember the explanation with trains? 🚂 🚋 🚋 
Here you use `last` which each time walks the whole list, which is slow.

That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go

Let me know if you need any help with such implementation! 👍🏼